### DISCUSSION TITLE
Fully RTL Support

### DISCUSSION THREAD

**Discussion by logseq-io** [athensresearch/athens#935] [Feature Requests]
**Linked issue #250**

Athens: no RTL support
logseq: no RTL support
obsidian: no official RTL support, only plugin 
Roam Research: no RTL support, some css 

Just to put things in perspective: 
RTL (Arabic: 422 million speakers + Hebrew 5 million people + Persian/Farsi 110 million people + Urdu 70 million people )
I don't need to do the math here. ;)

PLEASE, remember that not everyone speak or write in English so please consider supporting RTL.

**Comment by logseq-io** [comment]
![Image of Yaktocat](https://cdn.discordapp.com/attachments/708124073544122399/830263171167420416/a7c7dff704b8d89e.jpg)
I'm not a designer but I tried to see how would Athens in Arabic would fits in a logo.
If you like the design then share your opinion.

**Comment by shanberg** [comment]
I think this can be separated into at least a few pretty tractible tasks:

- #949 Minor CSS refactor: logical positioning styles: "left" -> "inline-start", "top" -> "block-start", etc.
- LTR/RTL language detection and setting on the HTML element so the CSS applies correctly
- Refactor the keyboard shortcuts to account for lang direction (e.g. ltr: "right" at end of block goes down, where in rtl "right" should go up)
- Testing with RTL users to identify any resulting issues or further improvements

**Reply by logseq-io** [reply]
Thank you for providing these tractable tasks. 
I'm a normal user with no technical background, I can provide feedback as a RTL user.  
Regarding this task:
**LTR/RTL language detection and setting on the HTML element so the CSS applies correctly** 
as an RTL user I also mix LTR with RTL so the alignment is very important.

**Reply by shanberg** [reply]
That's helpful context, thanks @logseq-io.

Do you have any apps in mind that are good examples of handling multiple text directions?

**Comment by logseq-io** [comment]
Dynalist is a good example. 
![Image of Dynalist](https://i2.paste.pics/f844a77651c1de409a722d6a2fbbc565.png)

**Comment by Storm-Engineer** [comment]
A note about internalization:

Shortcuts shouldn't use keys that don't exist on International keyboards. For example, `\` does not exist as a dedicated key on Hungarian keyboards (we type it via `Alt Gr + Q`) making current sidebar shortcuts inaccessible.

**Comment by Limezy** [comment]
Inside a block, I kind of see the point : if the first word is RTL, then any LTR will be "encapsulated" in a giant RTL sentence (or the other way around). However, It's much harder for me to imagine what things would look like within a page, or within a block that has some sub-blocks. Imagine a page where one block out of two is RTL / LTR. That would make one bullet on the left, the next one on the right and so on... Messy no ?

**Reply by Limezy** [reply]
Sounds like the only way is to have one graph-level setting, that will set all bullets to the right, or all bullets to the left.
And then, even if the graph-level setting is RTL, support for LTR langages encapsulated inside RTL blocks, and the other way around.

**Reply by Limezy** [reply]
OK so after double checking, this is already the way it works right now
![image](https://user-images.githubusercontent.com/24638389/116086394-59828100-a6d2-11eb-9981-1ad0bac9bfa3.png)
